website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,518 @@
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') != 'CI' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+</File>
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object               |
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Developer 1 | dev           | dbt_dev1      |dev.marketing.dbt_dev1_my_model |
+| Developer 2 | dev           | dbt_dev2      |dev.marketing.dbt_dev2_my_model  |
+| CI PR 123   | ci            | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model|
+| CI PR 234   | ci            | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model|
+| Production  | prod          | analytics     |prod.marketing.my_model         |
+
+
+### 3. Using branch name as schema prefix
+
+For teams who prefer to isolate work based on the feature branch, you may want to take advantage of the `DBT_CLOUD_GIT_BRANCH` special environment variable. Please note that developers will write to the exact same schema when they are on the same feature branch.
+
+
+:::note
+
+The `DBT_CLOUD_GIT_BRANCH` variable is only available within the dbt Cloud IDE and not the Cloud CLI.
+
+:::
+
+
+We’ve also seen some organizations prefer to organize their dev databases by branch name. This requires implementing similar logic in `generate_database_name()` instead of the `generate_schema_name()` macro. By default, dbt will not automatically create the databases. 
+
+Refer to the [Tips and tricks](p/guides/customize-schema-alias?step=5) section on the next page to learn more.    
+
+
+`<File name='macros/generate_schema_name.sql'>`
+
+```jinja
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if  env_var('DBT_ENV_TYPE','DEV') == 'DEV' -%}
+    
+        {#- we replace characters not allowed in the schema names by "_" -#}
+        {%- set re = modules.re -%}
+        {%- set cleaned_branch = re.sub("\W", "_", env_var('DBT_CLOUD_GIT_BRANCH')) -%}
+        
+        {%- if custom_schema_name is none -%}
+
+            {{ cleaned_branch }}
+
+        {%- else -%}
+
+             {{ cleaned_branch }}_{{ custom_schema_name | trim }}
+
+        {%- endif -%}
+        
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
```

Comment:
```suggestion

</File>

This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
```